src/draw/grouping.typ

```diff
@@ -432,13 +453,14 @@
 /// ```)
 ///
 /// - layer (float, integer): The layer to place the elements on. Elements placed without `on-layer` are always placed on layer 0.
-/// - body (elements): Elements to draw on the layer specified.
+/// - body (elements, function): Elements to draw on the layer specified. A function that accepts `ctx` and returns elements is also accepted.
 #let on-layer(layer, body) = {
-  assert(type(layer) in (int, float), message: "Layer must be a float or integer, 0 being the default layer. Got: " + repr(layer))
-  assert(type(body) in (function, array))
```

Comment:
Keep the layer assert here so its easier for the user to find. It might be worth checking the body anyway for the same reason? Then if all elements check that before calling the resolver, the resolver only has to check the output of the function...

Reply:
The assert is still there.

Reply:
Checking the body for what?

Reply:
So it is, bad github diff. I mean check that body is a function, array or none outside of the element function
```
assert(type(layer) ...)
assert(type(body) ...)
return ctx => ...
```
Then the error message will come from the user code and not hidden in all the function calls.

Reply:
I've added `util.assert-body(...)` for early checking of the body argument(s).